common/poller/AbstractPoller.java

```diff
@@ -18,21 +18,27 @@
 
 package com.vaticle.typedb.core.common.poller;
 
-import com.vaticle.typedb.core.common.iterator.FunctionalIterator;
-
 import java.util.Optional;
 import java.util.function.Function;
 
-public class AbstractPoller<T> implements Poller<T> {
+public abstract class AbstractPoller<T> implements Poller<T> {
 
     @Override
     public Optional<T> poll() {
         return Optional.empty();
     }
 
     @Override
-    public <U> Poller<U> flatMap(Function<T, FunctionalIterator<U>> flatMappingFn) {
+    public <U> Poller<U> flatMap(Function<T, Poller<U>> flatMappingFn) {
         return new FlatMappedPoller<>(this, flatMappingFn);
     }
 
+    @Override
+    public Poller<T> link(Poller<T> toLink) {
```

Comment:
you should match the naming conventions of the normal `Iterators` - the name here would then just be `poller` for the arg